napari_laptrack/_function.py

```diff
@@ -8,6 +8,27 @@
 def napari_experimental_provide_function():  # noqa: D103
     return [track_labels_centroid_based]
 
+def _check_and_convert_layers_to_4d(image_layer,labels_layer,viewer):
+    import napari_time_slicer as nts
+    image = image_layer.data
+    labels = labels_layer.data
+    # Deal with input data of various formats and bring both in [T,Z,Y,X] shape/format, maybe with Z=1
+    assert image.shape == labels.shape, "Image and labels must have the same shape."    
+    if len(image) == 2:
+        print("Image data seems 2D/on-the-fly-processed, converting to 3D+t...")
+        image_layer_4d = nts._function.convert_to_stack4d(image_layer, viewer)
+        labels_layer_4d = nts._function.convert_to_stack4d(labels_layer, viewer)
+    elif len(image_layer.data.shape) == 3:
+        print("Image data seems 2D+t, converting to 3D+t...")
+        image_layer_4d = nts._function.convert_to_2d_timelapse(image_layer, viewer)
+        labels_layer_4d = nts._function.convert_to_2d_timelapse(labels_layer, viewer)
```

Comment:
We might need to write that a little differently: What if image is 3d and labels is 4d? What if both are already 4d?

Shall we write a function approximately like this?
```
def _check_and_convert_image_of_any_kind():
    if len(image.shape) == 2:
        return nts._function.convert_to_stack4d(image_layer, viewer)
    elif len(image.shape) == 3:
        return nts._function.convert_to_2d_timelapse(image_layer, viewer)
    elif len(image.shape) == 4:
        return image
    else:
        raise ValueError(f"image shape {image.shape} not supported")
```

Reply:
Thanks! I rewrote this part so that they treat images and labels separately. I excluded the 2D case. Does it look fine?
https://github.com/haesleinhuepf/napari-laptrack/blob/175130b891a40209a849852eb988a647e7be26a9/napari_laptrack/_function.py#L11-L19